org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/dropwizard/xtext/validation/ValidationMarkerUpdater.xtend

```diff
@@ -0,0 +1,110 @@
+package org.testeditor.web.dropwizard.xtext.validation
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.ArrayList
+import java.util.HashMap
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.eclipse.emf.common.util.URI
+import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.xtext.builder.standalone.IIssueHandler
+import org.eclipse.xtext.validation.Issue
+import org.slf4j.LoggerFactory
+
+
+/**
+ * Collects and summarizes validation issues on a per-resource-basis.
+ * 
+ * This class is an {@link org.eclipse.xtext.builder.standalone.IIssueHandler},
+ * so when bound via Guice, it will be used by the {@link org.eclipse.xtext.builder.standalone.StandaloneBuilder}
+ * as a callback invoked after validating each resource.
+ * 
+ * This updater in turn sends the summarized validation results to a
+ * {@link org.testeditor.web.dropwizard.xtext.validation.ValidationMarkerMap},
+ * but only after the builder has finished validating all resources. Since the
+ * latter does not offer appropriate callback hooks, 
+ * {@link org.testeditor.web.xtext.index.CustomStandaloneBuilder} extends that
+ * class, and calls this updater's {@link #updateMarkerMap) after the build is
+ * complete.
+ * 
+ * While the mode of usage by the builder means that all issues passed to 
+ * {@link #handleIssue} will refer to the same resource, the method is, in
+ * theory, able to handle a mixed set of issues belonging to arbitrary
+ * resources. Unfortunately, not all validators produce issues with a non-null
+ * resource URI set, so as a fallback, the context resource can be explicitly
+ * set with {@link #setContext}, and its URI will be used to construct a
+ * resource path string for any issue that lacks the necessary information
+ * itself. 
+ */
+@Singleton
+class ValidationMarkerUpdater implements IIssueHandler {
+
+	@Inject DefaultIssueHandler delegateForDefaultBehavior
+	@Inject extension ValidationMarkerMap validationMarkerMap
+
+	static val logger = LoggerFactory.getLogger(ValidationMarkerUpdater)
+
+	val collectedValidationSummaries = new HashMap<String, ValidationSummary>()
+	var Path rootDirectory
+	var Resource currentResource
+
+	def void init(String rootDirectory) {
+		this.rootDirectory = Paths.get(rootDirectory)
+	}
+
+	def setContext(Resource resource) {
+		currentResource = resource
+	}
+
+	override handleIssue(Iterable<Issue> issues) {
+		if (issues !== null && issues.length > 0) {
+			issues.forEach [
```

Comment:
issues?.forEach should be enough (but is null really a value that is used here?)

Reply:
`null` will probably not be an issue, ever, but on the off-chance that it does occur (this is called from within Xtext framework code), I don't want a NPE to bubble up here.